util/buildkitskipper/hasher/hasher.go

```diff
@@ -30,6 +31,16 @@ func (h *Hasher) GetHash() []byte {
 	return h.h.Sum(nil)
 }
 
+func (h *Hasher) HashStatement(name string, args []string, body spec.Block) {
```

Comment:
This function seems very specific in requiring a single body, I think we should rename it to reflect that it's not a general use function, maybe just call it:

```suggestion
func (h *Hasher) HashWaitStatement(waitStmt spec.WaitStatement) {
```

honestly pushing down the `spec` into the hasher package feels like a wrong direction, which goes back to the original thread that @vladaionescu started.

The fact that we're passing in a `spec.*` yet only processing half of it in this package, and the other half in the `inputgraph` package feels like we have incorrectly broken this logic up into too many packages.

Reply:
ultimately I think the `HashCommand`, `HashVersion`, and now `HashWaitStatement` all belong in the `inputgraph` package.

perhaps this package could contain a helper function like `HashJSONMarshalled` which would take `v any`.

Reply:
Yes, that's pretty much why I started with the WAIT-specific logic in `inputgraph.go`. It seems less desirable to have `Hasher` become aware of the specific `spec` types. Updated with suggestions.